code/datums/brain_damage/brain_trauma.dm

```diff
@@ -41,16 +41,18 @@
 /datum/brain_trauma/proc/on_gain()
 	to_chat(owner, gain_text)
 	RegisterSignal(owner, COMSIG_MOB_SAY, .proc/handle_speech)
+	RegisterSignal(owner, COMSIG_MOVABLE_HEAR, .proc/handle_hearing)
 
 //Called when removed from a mob
 /datum/brain_trauma/proc/on_lose(silent)
 	if(!silent)
 		to_chat(owner, lose_text)
 	UnregisterSignal(owner, COMSIG_MOB_SAY)
+	UnregisterSignal(owner, COMSIG_MOVABLE_HEAR)
 
 //Called when hearing a spoken message
-/datum/brain_trauma/proc/on_hear(message, speaker, message_language, raw_message, radio_freq)
-	return message
+/datum/brain_trauma/proc/handle_hearing(datum/source, list/hearing_args)
+	UnregisterSignal(owner, COMSIG_MOVABLE_HEAR)
```

Comment:
nie ogarniam troche, z tego co rozumiem, brain trauma rejestruje sygnał proca który calluje sie po usłyszeniu rozmowy, ale sam ten proc odrazu usuwa ten sygnał, w takim razie po co ten sygnał?

Reply:
portowałem to z tg, więc nie wiem dokładnie co autor miał na myśli, ale zgaduje że to coś na wszelki wypadek gdyby jakaś choroba, która nie powinna tego wykorzystywać z jakiegoś powodu to wykorzystywała. wszystkie traumy, które faktycznie używają tego sygnału, nadpisują tą funkcje